### PR TITLE
#138 table of contents (Python/Ruby/Rust/Scala/Swift)

### PR DESCRIPTION
**Pull request by iskandarzulkarnaien**
Adds table of contents (toc) to articles without one and hides them except for mobile and print view as discussed in #138 . 

This PR affects the below chapters under `Programming Languages`. Kindly use the below checklist for reviewing.

Items to test for each chapter:
- [ ] Web view (Expected: in-article toc absent)
- [ ] Mobile view (Expected: in-article toc present and **links clickable**, right nav panel absent)
- [ ] Print view (Expected: in-article toc present, left and right nav panels absent)

Chapters:
- [ ] Python
- [ ] Ruby
- [ ] Rust
- [ ] Scala
- [ ] Swift

Devtools shortcuts
Firefox:
<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>M</kbd>

Chrome:
<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>I</kbd>

### CODE REVIEW

File: css/main.css [hunk 1]
```diff
@@ -27,3 +27,14 @@ mark {
 .indented-level3 {
     padding-left: 60px;
 }
+
+/* Responsive table of contents (toc) */
+
+/* Hides toc on screen widths smaller than 1300px */
+/* page-nav appears on screens 1300px wide and above */
+@media screen and (min-width: 1299.98px) {
```
Comment (madanalogy) [inline css/main.css:35]: Did you have any particular reason for not using 1300px? I'm thinking it might work more smoothly with page-nav because now there's a little sweet spot where neither are shown:

Screenshot 2020-02-21 at 9 47 46 AM

**Review by madanalogy — COMMENTED** [review]
Clean implementation, just a comment regarding threshold selection :)

File: css/main.css (hunk 1 above)
Comment (iskandarzulkarnaien) [reply css/main.css:35]: I wanted to be consistent with `page-nav.css` which has the following:
```
/* Responsive page navigation */

/* Hides page navigation on screen widths smaller than 1300px */
@media screen and (max-width: 1299.98px) {
    #page-nav {
        border: none;
        overflow: hidden;
        padding: 0;
        width: 0;
    }
}
```

But I'm curious, how did you manage to get both the toc and page-nav to disappear?

Comment (madanalogy) [reply css/main.css:35]: > I wanted to be consistent with `page-nav.css` which has the following:

Ah okay that makes sense, thanks for the clarification.

> But I'm curious, how did you manage to get both the toc and page-nav to disappear?

I just messed around with screen widths haha, but honestly it's an edge case and not expected user behaviour so my personal opinion is that it should be fine

Feb-21-2020 15-01-55

Comment (iskandarzulkarnaien) [reply css/main.css:35]: > I just messed around with screen widths haha, but honestly it's an edge case and not expected user behaviour so my personal opinion is that it should be fine

Ah I see. I manually inputted 1299 and 1300 into the dev tools to test since it only accepts integer values.

I agree that it is just an edge case, most users won't encounter this

**Comment by iskandarzulkarnaien** [comment]
I've removed the redundant first link of every toc as per @dvrylc 's suggestion

**Comment by iskandarzulkarnaien** [comment]
@madanalogy could you approve the PR if there are no changes, thanks!
@anubh-v could you review if you have the time, thanks!

**Review by anubh-v — APPROVED** [review]
Tested on Chrome version 80, works as intended  :+1:

**Review by madanalogy — APPROVED** [review]
LGTM!

**Comment by iskandarzulkarnaien** [comment]
Thanks @madanalogy and @anubh-v !
@damithc my PR is ready to merge, thanks!
